go/vt/vtgate/planbuilder/operators/ast2op.go

```diff
@@ -186,22 +208,243 @@ func createOperatorFromUpdate(ctx *plancontext.PlanningContext, updStmt *sqlpars
 	return subq, nil
 }
 
-// ColumnModified checks if any column in the parent table is being updated which has a child foreign key.
-func ColumnModified(exprs sqlparser.UpdateExprs, getFks func(expr *sqlparser.UpdateExpr) ([]vindexes.ParentFKInfo, []vindexes.ChildFKInfo)) bool {
-	for _, updateExpr := range exprs {
-		parentFKs, childFks := getFks(updateExpr)
-		for _, childFk := range childFks {
+// getFKRequirementsForUpdate analyzes update expressions to determine which foreign key constraints needs management at the VTGate.
+// It identifies parent and child foreign keys that require verification or cascade operations due to column updates.
+func getFKRequirementsForUpdate(updateExprs sqlparser.UpdateExprs, vindexTable *vindexes.Table) ([]vindexes.ParentFKInfo, []vindexes.ChildFKInfo) {
+	parentFks := vindexTable.ParentFKsNeedsHandling()
+	childFks := vindexTable.ChildFKsNeedsHandling(vindexes.UpdateAction)
+	if len(childFks) == 0 && len(parentFks) == 0 {
+		return nil, nil
+	}
+
+	pFksRequired := make([]bool, len(parentFks))
+	cFksRequired := make([]bool, len(childFks))
+	// Go over all the update expressions
+	for _, updateExpr := range updateExprs {
+		// Any foreign key to a child table for a column that has been updated
+		// will require the cascade operations to happen, so we include all such foreign keys.
+		for idx, childFk := range childFks {
 			if childFk.ParentColumns.FindColumn(updateExpr.Name.Name) >= 0 {
-				return true
+				cFksRequired[idx] = true
 			}
 		}
-		for _, parentFk := range parentFKs {
+		// If we are setting a column to NULL, then we don't need to verify the existance of an
+		// equivalent row in the parent table, even if this column was part of a foreign key to a parent table.
+		_, isNull := updateExpr.Expr.(*sqlparser.NullVal)
+		if isNull {
+			continue
+		}
+		// We add all the possible parent foreign key constraints that need verification that an equivalent row
+		// exists, given that this column has changed.
+		for idx, parentFk := range parentFks {
 			if parentFk.ChildColumns.FindColumn(updateExpr.Name.Name) >= 0 {
-				return true
+				pFksRequired[idx] = true
 			}
 		}
 	}
-	return false
+	// For the parent foreign keys, if any of the columns part of the fk is set to NULL,
+	// then, we don't care for the existance of an equivalent row in the parent table.
+	for idx, parentFk := range parentFks {
+		for _, updateExpr := range updateExprs {
+			_, isNull := updateExpr.Expr.(*sqlparser.NullVal)
```

Comment:
nitpick: seems we are checking for NullVal in a few places. I think adding a method to `ast_func.go` that checks this and returns true or false would read nicely. 

```go
func IsNull((e Expr) bool {
  _, isNull := e.Expr.(*NullVal)
  return isNull
}
```